src/components/forms/checkbox.stories.mdx

```diff
@@ -0,0 +1,73 @@
+import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
+import { withKnobs, text, boolean } from '@storybook/addon-knobs';
+
+import Checkbox from './Checkbox';
+
+<Meta title="Forms|Checkbox" decorators={[withKnobs]} component={Checkbox} />
+
+# Checkbox
+
+Renders an FDS Icon in place of the user agent checkbox. The user agent `input
+type="checkbox"` is visually hidden behind the icon.
+
+The `name` prop is required, as it is used to create a reference between the `label` and
+`input` elements.
+
+<hr />
+
+<Story name="Knobs">
+  <Checkbox
+    indeterminate={boolean('indeterminate', false)}
+    disabled={boolean('disabled', false)}
+    name="default-checked"
+    label={text('label', 'Checkbox label')}
+  />
+</Story>
+
+<hr />
+
+<Props of={Checkbox} />
+
+## Managing checked state
+
+Just like a native `input type="checkbox"`, this component will render an unchecked box by
+default. Use the `checked` prop to render the checkbox as checked.
+
+**Users can still uncheck the box.** To "freeze" checked state, use a disabled checkbox
+(see below).
+
+<Preview>
+  <Story name="Managing checked state">
+    <Checkbox checked name="default-checked" label="Checked by default" />
+  </Story>
+</Preview>
+
+## Preventing user interaction
+
+Use the `disabled` prop to prevent the user from changing the chekced state of the
+checkbox.
+
+<Preview>
+  <Story name="Disabled checkboxes">
+    <React.Fragment>
+      <div>
+        <Checkbox disabled name="disabled-unchecked" label="Disabled unchecked" />
+      </div>
+      <div>
+        <Checkbox disabled checked name="disabled-checked" label="Disabled checked" />
+      </div>
+    </React.Fragment>
+  </Story>
+</Preview>
+
+## Indeterminate checkbox
+
+HTML checkboxes [natively support an indeterminate state](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#Using_checkbox_inputs).
```

Comment:
Again, I sort of wonder here if this is actually helpful for consumers. This seems like something that would go in the playbook (sans the documentation). I actually have a playbook section in the FDS v7 organizational PR

Reply:
I think you're right - this specific example probably isn't useful to many people (if any).

Reply:
A few thoughts on explanatory content in these docs generally...

I think you're right that the majority of users will only care about basic API docs for a component. There will be some users (maybe product, design, or curious engineers) that want to get a better understanding of the intention of a component or how it works.

The thing I love most about docblocks is that you're able to serve varying personas:

> I just need the API docs

look at the props table, skip the rest

> I'm doing something advanced with the component

read more of the examples in the docs (mostly `Preview` stuff)

> Why is this prop required?

read the explanations to get a better understanding

> I learn better by gathering more context

read the extra text for context

With all that said, I don't think this note about native support for indeterminate checkboxes is very useful. I only added that in response to a few misunderstandings about indeterminate states I heard from designers and engineers last month